src/Serialization/main.jl

```diff
@@ -130,12 +135,19 @@ function load_type_dispatch(s::DeserializerState, ::Type{T}, dict::Dict) where T
 
     # TODO: compare T against decodedType ???
     #decodedType = decodeType(dict[:type])
-
-    result = load_internal(s, T, dict[:data])
     id = dict[:id]
+    
     if id != "-1"
-        s.objs[UUID(id)] = result
+        if UUID(id) in keys(s.objs)
+            result = s.objs[UUID(id)]
```

Comment:
I don't understand. How could this line ever be triggered? If the id is already in there, doesn't that mean that we are now deserializing a second object with the *same* UUID, which would mean our data is corrupt? What am I missing?

Note that backrefs are different and handled above.

Reply:
didn't realize this wasn't handling backrefs, thanks